contracts/value/mod.go

```diff
@@ -0,0 +1,245 @@
+// Package value implements a simple native contract that can store, delete, and
+// display values.
+package value
+
+import (
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+
+	"go.dedis.ch/dela"
+	"go.dedis.ch/dela/core/access"
+	"go.dedis.ch/dela/core/execution"
+	"go.dedis.ch/dela/core/execution/native"
+	"go.dedis.ch/dela/core/store"
+	"golang.org/x/xerrors"
+)
+
+// commands defines the commands of the value contract. This interface helps in
+// testing the contract.
+type commands interface {
+	set(snap store.Snapshot, step execution.Step) error
+	read(snap store.Snapshot, step execution.Step) error
+	delete(snap store.Snapshot, step execution.Step) error
+	display(snap store.Snapshot) error
+}
+
+const (
+	// ContractName is the name of the contract.
+	ContractName = "go.dedis.ch/dela.Value"
+
+	// KeyArg is the argument's name in the transaction that contains the
+	// provided key to update.
+	KeyArg = "value:key"
+
+	// ValueArg is the argument's name in the transaction that contains the
+	// provided value to set.
+	ValueArg = "value:value"
+
+	// CmdArg is the argument's name to indicate the kind of command we want to
+	// run on the contract. Should be one of the Command type.
+	CmdArg = "value:command"
+)
+
+// Command defines a type of command for the value contract
+type Command string
+
+const (
+	// CmdSet defines the command to set a value
+	CmdSet Command = "SET"
+
+	// CmdRead defines a command to read a value
+	CmdRead Command = "READ"
+
+	// CmdDelete defines a command to delete a value
+	CmdDelete Command = "DELETE"
+
+	// CmdDisplay defines a command to display all values set (and not deleted)
+	// so far.
+	CmdDisplay Command = "DISPLAY"
+)
+
+// NewCreds creates new credentials for a value contract execution. We might
+// want to use in the future a separate credential for each command.
+func NewCreds(id []byte) access.Credential {
+	return access.NewContractCreds(id, ContractName, "all")
```

Comment:
what's `"all"` ? I'd consider putting that in a variable just to document the intention.

Reply:
that's the credential name, we could use different credentials for each action, ie. one for READ, WRITE, DISPLAY, so that's where we would use a different name. There we only have one global credential call "all".